addons/mail/models/mail_activity_log.py

```diff
@@ -0,0 +1,159 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from datetime import date, datetime, timedelta
+
+from odoo import api, fields, models, _
+from odoo.tools import DEFAULT_SERVER_DATE_FORMAT
+
+
+class MailActivity(models.Model):
+    _name = 'mail.activity'
+    _description = 'Mail Activity'
+    _inherits = {'mail.message.subtype': 'subtype_id'}
+    _rec_name = 'name'
+    _order = "sequence"
+
+    days = fields.Integer('Number of days', default=0,
+                          help='Number of days before executing the action, allowing you to plan the date of the action.')
+    subtype_id = fields.Many2one('mail.message.subtype', string='Message Subtype', required=True, ondelete='cascade')
+    icon = fields.Char(help="Font awesome icon. eg. fa-tasks", groups="base.group_no_one")
+    model_id = fields.Many2one('ir.model', string='Related Document Model', groups="base.group_no_one")
+
+    # setting a default value on inherited fields is a bit involved
+    internal = fields.Boolean('Internal Only', related='subtype_id.internal', inherited=True, default=True)
+    default = fields.Boolean('Default', related='subtype_id.default', inherited=True, default=False)
+
+
+class MailActivityLog(models.Model):
+    _name = "mail.activity.log"
+    _description = "Log an Activity"
+    _order = "create_date desc"
+    _rec_name = 'title_action'
+
+    # If date_action is more than DUE_DAYS days, display the DATE
+    # else `Due in X days`
+    DUE_DAYS = 15
```

Comment:
i am not sure why this variable required. ? because it's only used inside one function. 
can not we directly define one variable inside that function fetch_activity_log.?